canonical-web-and-design/dotrun
# Pip support

Support for detecting and installing Python dependencies from `requirements.txt` with Pip3 into a `.venv` python virtual env directory.

QA
--

Install dotrun as explained [in the README](https://github.com/nottrobin/dotrun/tree/python#installation). Then:

``` bash
../ubuntu.com  # Switch to a clone of the ubuntu.com repo
echo "gunicorn==20.0.4" >> requirements.txt  # strict confinement only works with gunicorn >= 20
dotrun build
dotrun serve
```

- See that it serves the site successfully
- Cancel the server
- Notice that `.dotrun.json` and `.venv` were created
- Do `dotrun serve` again, check it doesn't install dependencies this time
- Change something in `requirements.txt`, and run `dotrun serve` and see that it *does* reinstall dependencies
- Do `dotrun clean` and see that `.dotrun.json` and `.venv` are cleaned up
